internal/inputs/http_test.go

```diff
@@ -7,102 +7,318 @@ package inputs
+					"api.StatusCode":            200,
+					"api.header.Content-Type":   []string{"application/json"},
+					"api.header.Content-Length": []string{"216"},
+					"api.header.Date":           []string{"Mon, 18 May 2020 09:38:35 GMT"},
+					"api.header.Retry-Count":    []string{"0"},
+				},
+				map[string]interface{}{
+					"userId":                    float64(1),
+					"id":                        float64(2),
+					"title":                     "quis ut nam facilis et officia qui",
+					"completed":                 "false",
+					"api.StatusCode":            200,
+					"api.header.Content-Type":   []string{"application/json"},
+					"api.header.Content-Length": []string{"216"},
+					"api.header.Date":           []string{"Mon, 18 May 2020 09:38:35 GMT"},
+					"api.header.Retry-Count":    []string{"0"},
 				},
 			},
-			{
-				EventType: "httpExample3",
-				URL:       "users",
-				Timeout:   5100,
-				User:      "batman",
-				Pass:      "robin",
-				Headers: map[string]string{
-					"test2": "abc2",
+			"../../test/payloadsExpected/http_response/http_response-multiple_objects.json",
+		},
+		"sample-with-headers-string-response": {
+			"127.0.0.1:9126",
+			load.Config{
+				Name: "return-headers-example",
+				Global: load.Global{
+					BaseURL: "http://127.0.0.1:9126",
+				},
+				APIs: []load.API{
+					{
+						EventType:     "return-headers-example",
+						URL:           "/",
+						Timeout:       5100,
+						ReturnHeaders: true,
+					},
 				},
 			},
+			[]interface{}{
+				map[string]interface{}{
+					"output":                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
+					"api.StatusCode":            200,
+					"api.header.Content-Type":   []string{"application/json"},
+					"api.header.Content-Length": []string{"136"},
+					"api.header.Date":           []string{"Mon, 18 May 2020 09:38:35 GMT"},
+					"api.header.Retry-Count":    []string{"0"},
+				},
+			},
+			"../../test/payloadsExpected/http_response/http_response-string_response.json",
 		},
-	}
-
-	expectedSamples := []interface{}{
-		map[string]interface{}{
-			"userId":         1,
-			"id":             1,
-			"title":          "delectus aut autem",
-			"completed":      "false",
-			"api.StatusCode": 200,
+		"sample-error-with-headers": {
+			"127.0.0.1:9127",
+			load.Config{
+				Name: "return-headers-example",
+				Global: load.Global{
+					BaseURL: "http://127.0.0.1:9127",
+				},
+				APIs: []load.API{
+					{
+						EventType:     "return-headers-example",
+						URL:           "/",
+						Timeout:       5100,
+						ReturnHeaders: true,
+					},
+					{
+						EventType:     "return-headers-example",
+						URL:           "/todos",
+						Timeout:       5100,
+						ReturnHeaders: true,
+					},
+				},
+			},
+			[]interface{}{
+				map[string]interface{}{
+					"error":                     "Missing Required Parameters",
+					"api.StatusCode":            200,
```

Comment:
this is strange. If statusCode is 200, "error" should not be present. I think error is only when the http call fails, no?

Reply:
Yes, totally. As an expected value could be more real with a 5xx status code. Will change it.

However I've saw apis that return success status code with error message in body XD